pkg/scheduler/framework/plugins/defaultpodtopologyspread/default_pod_topology_spread.go

```diff
@@ -86,3 +95,141 @@ func New(_ *runtime.Unknown, handle framework.FrameworkHandle) (framework.Plugin
 		calculateSpreadPriorityReduce: calculateSpreadPriorityReduce,
 	}, nil
 }
+
+//Migration Code
+
+type SelectorSpread struct {
+	serviceLister     corelisters.ServiceLister
+	controllerLister  corelisters.ReplicationControllerLister
+	replicaSetLister  appslisters.ReplicaSetLister
+	statefulSetLister appslisters.StatefulSetLister
+}
+
+// NewSelectorSpreadPriority creates a SelectorSpread.
+func NewSelectorSpreadPriority(
+	serviceLister corelisters.ServiceLister,
+	controllerLister corelisters.ReplicationControllerLister,
+	replicaSetLister appslisters.ReplicaSetLister,
+	statefulSetLister appslisters.StatefulSetLister) (priorities.PriorityMapFunction, priorities.PriorityReduceFunction) {
+	selectorSpread := &SelectorSpread{
+		serviceLister:     serviceLister,
+		controllerLister:  controllerLister,
+		replicaSetLister:  replicaSetLister,
+		statefulSetLister: statefulSetLister,
+	}
+	return selectorSpread.CalculateSpreadPriorityMap, selectorSpread.CalculateSpreadPriorityReduce
+}
+
+// CalculateSpreadPriorityMap spreads pods across hosts, considering pods
+// belonging to the same service,RC,RS or StatefulSet.
+// When a pod is scheduled, it looks for services, RCs,RSs and StatefulSets that match the pod,
+// then finds existing pods that match those selectors.
+// It favors nodes that have fewer existing matching pods.
+// i.e. it pushes the scheduler towards a node where there's the smallest number of
+// pods which match the same service, RC,RSs or StatefulSets selectors as the pod being scheduled.
+func (s *SelectorSpread) CalculateSpreadPriorityMap(pod *v1.Pod, meta interface{}, nodeInfo *schedulernodeinfo.NodeInfo) (framework.NodeScore, error) {
+	var selector labels.Selector
+	node := nodeInfo.Node()
+	if node == nil {
+		return framework.NodeScore{}, fmt.Errorf("node not found")
+	}
+
+	priorityMeta, ok := meta.(*priorityMetadata)
+	if ok {
+		selector = priorityMeta.podSelector
+	} else {
+		selector = getSelector(pod, s.serviceLister, s.controllerLister, s.replicaSetLister, s.statefulSetLister)
+	}
```

Comment:
this needs to be replaced with a PostFilter.

Reply:
see this plugin as an example: https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/plugins/tainttoleration/taint_toleration.go#L84

Reply:
Where should the PostFilter logic go? In this same file or in a different location?

Reply:
same file, just like the example I linked to.